src/csharp/Utils.cs

```diff
@@ -53,45 +54,56 @@ public static int GetCurrentGpuDeviceId()
 
         public static void SetLogBool(string name, bool value)
         {
-            Result.VerifySuccess(NativeMethods.OgaSetLogBool(StringUtils.ToUtf8(name), value));
+            Result.VerifySuccess(NativeMethods.OgaSetLogBool(StringUtils.ToNullTerminatedUtf8(name), value));
         }
         
         public static void SetLogString(string name, string value)
         {
-            Result.VerifySuccess(NativeMethods.OgaSetLogString(StringUtils.ToUtf8(name), StringUtils.ToUtf8(value)));
+            Result.VerifySuccess(NativeMethods.OgaSetLogString(StringUtils.ToNullTerminatedUtf8(name), StringUtils.ToNullTerminatedUtf8(value)));
         }
     }
 
     internal class StringUtils
     {
-        internal static byte[] EmptyByteArray = new byte[] { 0 };
+        internal static readonly byte[] EmptyByteArray = [0];
 
-        internal static byte[] ToUtf8(string str)
+        internal static byte[] ToNullTerminatedUtf8(string str) => ToNullTerminatedUtf8(str.AsSpan());
+
+        internal static unsafe byte[] ToNullTerminatedUtf8(ReadOnlySpan<char> str)
         {
-            if (string.IsNullOrEmpty(str))
+            if (str.IsEmpty)
                 return EmptyByteArray;
 
-            int arraySize = UTF8Encoding.UTF8.GetByteCount(str);
-            byte[] utf8Bytes = new byte[arraySize + 1];
-            UTF8Encoding.UTF8.GetBytes(str, 0, str.Length, utf8Bytes, 0);
-            utf8Bytes[utf8Bytes.Length - 1] = 0;
-            return utf8Bytes;
-        }
-
-        internal static string FromUtf8(IntPtr nativeUtf8)
-        {
-            unsafe
+            fixed (char* pStr = str)
             {
-                int len = 0;
-                while (*(byte*)(nativeUtf8 + len) != 0) ++len;
-
-                if (len == 0)
+                int byteCount = Encoding.UTF8.GetByteCount(pStr, str.Length);
```

Comment:
> int byteCount = Encoding.UTF8.GetByteCount(pStr, str.Length); [](http://example.com/codeflow?start=15&length=62)

Can use GetMaxCount instead with ArrayPool? just to avoid processing the text twice.